lib/ansible/modules/cloud/amazon/cloudfront_distribution.py

```diff
@@ -0,0 +1,2106 @@
+            return self.paginated_response(func)
+        except botocore.exceptions.ClientError as e:
+            self.module.fail_json(
+                msg="error deleting streaming distribution - " + str(e),
+                exception=traceback.format_exc(),
+                **camel_dict_to_snake_dict(e.response))
+
+    def update_streaming_distribution(self, config, streaming_distribution_id,
+                                      e_tag):
+        try:
+            func = partial(self.client.update_streaming_distribution,
+                           StreamingDistributionConfig=config,
+                           Id=streaming_distribution_id, IfMatch=e_tag)
+            return self.paginated_response(func)
+        except botocore.exceptions.ClientError as e:
+            self.module.fail_json(
+                msg="error updating streaming distribution - " + str(e),
+                exception=traceback.format_exc(),
+                **camel_dict_to_snake_dict(e.response))
+
+    def tag_resource(self, arn, tags):
+        try:
+            func = partial(self.client.tag_resource, Resource=arn,
+                           Tags=tags)
+            return self.paginated_response(func)
+        except botocore.exceptions.ClientError as e:
+            self.module.fail_json(
+                msg="error tagging resource - " + str(e),
+                exception=traceback.format_exc(),
+                **camel_dict_to_snake_dict(e.response))
+
+    def untag_resource(self, arn, tag_keys):
+        try:
+            func = partial(self.client.untag_resource, Resource=arn,
+                           TagKeys={'Items': tag_keys})
+            return self.paginated_response(func)
+        except botocore.exceptions.ClientError as e:
+            self.module.fail_json(
+                msg="error untagging resource - " + str(e),
+                exception=traceback.format_exc(),
+                **camel_dict_to_snake_dict(e.response))
+
+    def remove_all_tags_from_resource(self, arn):
+        tags = self.list_tags_for_resource(arn)
+        key_list = []
+        for tag in tags:
+            key_list.append(tag.get('Key'))
+        self.untag_resource(arn, key_list)
+
+    def list_tags_for_resource(self, arn):
+        try:
+            func = partial(self.client.list_tags_for_resource, Resource=arn)
+            response = self.paginated_response(func)
+            return response.get('Tags').get('Items')
+        except botocore.exceptions.ClientError as e:
+            self.module.fail_json(
+                msg="error removing all tags from resource - " + str(e),
+                exception=traceback.format_exc(),
+                **camel_dict_to_snake_dict(e.response))
+
+    def update_tags(self, valid_tags, purge_tags, arn, alias,
+                    distribution_id, streaming_distribution_id):
+        try:
+            changed = False
+            if purge_tags:
+                self.remove_all_tags_from_resource(arn)
```

Comment:
purge_tags doesn't need to remove all tags - you can use compare_aws_tags() (in module_utils/ec2) and then call untag_resource() with the list of tag_keys that are returned.